helios-client/src/main/java/com/spotify/helios/common/descriptors/RollingOperation.java

```diff
@@ -0,0 +1,247 @@
+      @JsonProperty("job") @Nullable final JobId jobId,
+      @JsonProperty("reason") @Nullable final Reason reason,
+      @JsonProperty("rolloutOptions") @Nullable final RolloutOptions rolloutOptions) {
+    this.id = id;
+    this.deploymentGroupName = deploymentGroupName;
+    this.jobId = jobId;
+    this.reason = reason;
+    this.rolloutOptions = rolloutOptions;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public String getDeploymentGroupName() {
+    return deploymentGroupName;
+  }
+
+  public JobId getJobId() {
+    return jobId;
+  }
+
+  public Reason getReason() {
+    return reason;
+  }
+
+  public RolloutOptions getRolloutOptions() {
+    return rolloutOptions;
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    final RollingOperation that = (RollingOperation) o;
+
+    if (jobId != null ? !jobId.equals(that.jobId) : that.jobId != null) {
+      return false;
+    }
+    if (deploymentGroupName != null ? !deploymentGroupName.equals(that.deploymentGroupName)
+                                    : that.deploymentGroupName != null) {
+      return false;
+    }
+    if (id != null ? !id.equals(that.id) : that.id != null) {
+      return false;
+    }
+    if (reason != null ? !reason.equals(that.reason) : that.reason != null) {
+      return false;
+    }
+    if (rolloutOptions != null ? !rolloutOptions.equals(that.rolloutOptions)
+                               : that.rolloutOptions != null) {
+      return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = id != null ? id.hashCode() : 0;
+    result = 31 * result + (deploymentGroupName != null ? deploymentGroupName.hashCode() : 0);
+    result = 31 * result + (jobId != null ? jobId.hashCode() : 0);
+    result = 31 * result + (reason != null ? reason.hashCode() : 0);
+    result = 31 * result + (rolloutOptions != null ? rolloutOptions.hashCode() : 0);
+    return result;
+  }
+
+  @Override
+  public String toString() {
+    return "RollingOperation{" +
+           "id='" + id + '\'' +
+           ", deploymentGroupName=" + deploymentGroupName +
+           ", job=" + jobId +
+           ", reason=" + reason +
+           ", rolloutOptions=" + rolloutOptions +
+           '}';
+  }
+
+  public Builder toBuilder() {
+    final Builder builder = newBuilder();
+
+    return builder
+        .setJobId(jobId)
+        .setDeploymentGroupName(deploymentGroupName)
+        .setReason(reason)
+        .setRolloutOptions(rolloutOptions);
+  }
+
+  public static class Builder implements Cloneable {
+
+    private final Parameters p;
+
+    private Builder() {
+      this.p = new Parameters();
+    }
+
+    private static class Parameters implements Cloneable {
```

Comment:
I don't actually know. I was talking about this with @mattnworb the other day - we seem to have Builders with both patterns. I used this pattern because I copy pasta'd from `DeploymentGroup`.